include/ck_tile/ops/reduce/pipeline/reduce2d_default_policy.hpp

```diff
@@ -76,5 +80,31 @@ struct Reduce2dDefaultPolicy
             return 1; // zero size arrays are an extension
         }
     }
+
+    template <typename Problem>
+    CK_TILE_HOST_DEVICE static constexpr index_t GetSmemSizeForIndices()
+    {
+        if constexpr(Problem::kNeedCrossWarpSync && Problem::kOutputIndex)
+        {
+            using P_ = BlockReduce2dProblem<typename Problem::XDataType,
+                                            typename Problem::ComputeDataType,
+                                            typename Problem::BlockShape,
+                                            Problem::kOutputIndex>;
+
+            using block_reduce2d = BlockReduce2d<P_>;
+            using x_block_tile =
+                decltype(make_static_distributed_tensor<typename Problem::XDataType>(
+                    MakeXBlockTileDistribution<Problem>()));
+            using y_index_block_tile =
+                decltype(block_reduce2d::template MakeYIndexBlockTile<x_block_tile, index_t>());
+
+            return GetBlockReduce2dCrossWarpSync<Problem>()
+                .template GetSmemSizeForIndices<y_index_block_tile>();
+        }
+        else
+        {
+            return 1; // zero size arrays are an extension
```

Comment:
It might be cumbersome but if you would have a simple function like `constexpr NeedIndicesSmem() -> bool` where you'd just put the condition `return Problem::kNeedCrossWarpSync && Problem::kOutputIndex` then you could guard smem allocation with it.